autoload/inline_edit.vim

```diff
@@ -172,3 +172,84 @@ function! inline_edit#HereDoc()
 
   return [start, end, filetype, indent]
 endfunction
+
+" function! inline_edit#PythonQuotedString() {{{2
+
+function s:check_inside_python_string()
+  return index(map(synstack(line('.'), col('.')), 'synIDattr(v:val, "name")'), "pythonString") >= 0
+endfunction
+" Opens up a new proxy buffer with the contents of a fenced code block in
+" github-flavoured markdown.
+function! inline_edit#PythonMultilineString()
+  call inline_edit#PushCursor()
+
+  normal! 0
+
+  if !s:check_inside_python_string()
+    call inline_edit#PopCursor()
+    return []
+  endif
+
+  normal! $
+
+  if !s:check_inside_python_string()
+    call inline_edit#PopCursor()
+    return []
+  endif
+
+  " We are inside a Python multiline string, so we have to find the boundaries
+  let quote_type = search('"""\|\(''''''\)', 'bWp')
+
+  if quote_type == 0
+
+    call inline_edit#PopCursor()
+    return []
+  endif
+
+  let start = line('.') + 1
+
+  if quote_type == 1
+    let end_quote = '"""'
+  elseif quote_type == 2
+    let end_quote = "'''"
+  else
+    echoerr "Invalid quote pattern found"
+    return []
+  endif
```

Comment:
This code confuses me, because the `quote_type` variable is the result of the `search()` call and that should be the line number of the match, and it doesn't seem to be used like this here. Could you explain how the `quote_type` variable takes the value of 1 or 2 depending on the match? Maybe I'm missing something in the variable assignments.

Reply:
According to `:help search()`, when the. `p` flag is used, the return value will be the something equivalent to the index of the match group inside the pattern. More specifically: `1` if the pattern matches but no subgroup matches, `2` if the first (and only) subgroup matched.

I use that to specifically identify whether `"""` (return value of `1`) or `'''` (return value of `2`) is being used in the multiline string. Afterwards it will try to find the matching triple multiline string mark.

Notice I didn't use `searchpair()` here because the `start` and `end` patterns are the same, and the function doesn't work in that case, as it assumes the end pattern to actually be a nested pair.

Reply:
Ah, I see -- I didn't know about the "p" flag or had forgotten about it, it makes a lot more sense now :+1:.